controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook_test.go

```diff
@@ -209,11 +214,17 @@ func TestKubeadmControlPlaneValidateUpdate(t *testing.T) {
 	invalidUpdateKubeadmConfigInit := before.DeepCopy()
 	invalidUpdateKubeadmConfigInit.Spec.KubeadmConfigSpec.InitConfiguration = &kubeadmv1beta1.InitConfiguration{}
 
+	validUpdateKubeadmConfigInit := before.DeepCopy()
+	validUpdateKubeadmConfigInit.Spec.KubeadmConfigSpec.InitConfiguration.NodeRegistration = kubeadmv1beta1.NodeRegistrationOptions{}
```

Comment:
Also, `before.InitConfiguration.NodeRegistration` is empty, so we not testing any change here 
https://github.com/kubernetes-sigs/cluster-api/blob/8a03ae7d608f6dfd3befa7b08ee076253d8b804e/controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook_test.go#L173-L178

Reply:
Added a `NodeRegistration` stanza. Thanks.